charmcraft/utils.py

```diff
@@ -185,3 +186,26 @@ def get_host_architecture():
     """Get host architecture in deb format suitable for base definition."""
     os_platform = get_os_platform()
     return ARCH_TRANSLATIONS.get(os_platform.machine, os_platform.machine)
+
+
+def confirm_with_user(prompt, default=False) -> bool:
+    """Query user for yes/no answer.
+
+    If TTY is not interaction, returns default value.
```

Comment:
Maybe "if stdin is not a terminal"?

Reply:
I must have bashed my head on keyboard :rofl:

Reply:
_interactive_ might be the word that is generic enough

Reply:
Improved docs!